core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -243,6 +224,7 @@ public TableScan ignoreResiduals() {
 
   @Override
   public CloseableIterable<CombinedScanTask> planTasks() {
+    ImmutableMap<String, String> options = context.options();
```

Comment:
We avoid leaking specific classes through the API. Can you update this to use `Map` instead? The getter method `options` should also only return `Map`.

Reply:
I guess this also applies to the already existing [BaseTableScan#options()](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseTableScan.java#L103).